src/components/Search/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React, {FC, useCallback, useRef, useState} from 'react';
+import { useDispatch } from 'react-redux';
+import {setSearchValue} from "../../redux/filter/slice";
+import debounce from 'lodash.debounce';
+
+
+const Search:FC = () => {
+    const dispatch=useDispatch;
+    const [value, setValue] = useState<string>('');
+    const inputRef=useRef<HTMLInputElement>(null)
+
+    const onClickClear=()=>{
+        // @ts-ignore
+        dispatch(setSearchValue(''));
+        setValue('')
+        inputRef.current?.focus();
+        
+    }
+
+    const updateSearchValue = React.useCallback(
```

Comment:
React.useCallback - in header component you imported hook in very different way , please make it consistent